validator/src/admin_rpc_service.rs

```diff
@@ -303,6 +327,111 @@ impl AdminRpc for AdminRpcImpl {
         AdminRpcImpl::set_identity_keypair(meta, identity_keypair, require_tower)
     }
 
+    fn set_block_engine_config(
+        &self,
+        meta: Self::Metadata,
+        block_engine_url: String,
+        trust_packets: bool,
+    ) -> Result<()> {
+        debug!("set_block_engine_config request received");
+
+        meta.with_post_init(|post_init| {
+            let mut auth_service_endpoint = Endpoint::from_shared(block_engine_url.clone())
+                .map_err(|_| {
+                    jsonrpc_core::error::Error::invalid_params(
+                        "invalid auth url value - block engine",
+                    )
+                })?;
+            if block_engine_url.contains("https") {
+                auth_service_endpoint = auth_service_endpoint
+                    .tls_config(tonic::transport::ClientTlsConfig::new())
+                    .map_err(|_| {
+                        jsonrpc_core::error::Error::invalid_params(
+                            "failed to set tls_config for block engine auth service",
+                        )
+                    })?;
+            }
+            let mut backend_endpoint = Endpoint::from_shared(block_engine_url.clone())
+                .map_err(|_| {
+                    jsonrpc_core::error::Error::invalid_params("invalid block engine url value")
+                })?
+                .tcp_keepalive(Some(Duration::from_secs(60)));
+
+            if block_engine_url.contains("https") {
+                backend_endpoint = backend_endpoint
+                    .tls_config(tonic::transport::ClientTlsConfig::new())
+                    .map_err(|_| {
+                        jsonrpc_core::error::Error::invalid_params(
+                            "failed to set tls_config for block engine service",
+                        )
+                    })?;
+            }
+
+            *post_init.block_engine_config.lock().unwrap() = BlockEngineConfig {
+                auth_service_endpoint: Some(auth_service_endpoint),
+                backend_endpoint: Some(backend_endpoint),
+                trust_packets,
+            };
+
+            Ok(())
+        })
+    }
+
+    fn set_relayer_config(
+        &self,
+        meta: Self::Metadata,
+        relayer_url: String,
+        trust_packets: bool,
+        expected_heartbeat_interval_ms: u64,
+        max_failed_heartbeats: u64,
+    ) -> Result<()> {
+        debug!("set_relayer_config request received");
+
+        meta.with_post_init(|post_init| {
+            let mut auth_service_endpoint =
+                Endpoint::from_shared(relayer_url.clone()).map_err(|_| {
+                    jsonrpc_core::error::Error::invalid_params("invalid auth url value - relayer")
```

Comment:
msg might be slightly confusing since the field being parsed is named `relayer_url`